uctMaths/competition/compadmin.py

```diff
@@ -1288,25 +1296,121 @@ def get_student_answer_sheet(request, student):
+            #Render the template with the context (from above)
+            template = get_template('printer_entry.html')
+            if request:
+                c.update(csrf(request))
+            context = Context(c)
+            register_html += template.render(context) #Concatenate each rendered template to the html "string"
+        else:
+            c = {'type':'Students',
+                'timestamp':timestamp,
+                'schooln': assigned_school,
+                'grades':range(8,13),
+                'grade_left':range(8,11),
+                'invigilator_range':range(10-len(invigilator_list)), 
+                'igrades':range(8,13),
+                'max_num_pairs':admin_number_of_pairs(),
+                'total_num':'No students entered for this school',
+                'invigilators_required':competition_has_invigilator()}
+
+            #Render the template with the context (from above)
+            template = get_template('printer_entry.html')
+            if request:
+                c.update(csrf(request))
+            context = Context(c)
+            register_html += template.render(context) #Concatenate each rendered template to the html "string"
+   
+    register_result = StringIO.StringIO()
+
+    #Generate the pdf doc
+    register_pdf = pisa.pisaDocument(StringIO.StringIO(register_html.encode("UTF-8")), register_result, encoding='UTF-8')
+    
+    if not register_pdf.err:
+        return register_result
+    else:
+        pass #Error handling?
+
 def printer_answer_sheet(request, assigned_school=None):
     """ Generate the school answer sheet for each school in the query set"""
-
-    html = '' #Will hold rendered templates
     student_list = SchoolStudent.objects.filter(school = assigned_school)
+    answer_sheets_result = StringIO.StringIO()
+    answer_sheets_html = ''
     for istudent in student_list:
-        html += get_student_answer_sheet(request, istudent) #Concatenate each rendered template to the html "string"
+        answer_sheets_html += get_student_answer_sheet(request, istudent)
 
-    result = StringIO.StringIO()
-    #Generate the pdf doc
-    pdf = pisa.pisaDocument(StringIO.StringIO(html.encode("UTF-8")), result, encoding='UTF-8')
-    if not pdf.err:
-        return result
+    if isinstance(assigned_school, list):
+        register_result = generate_answer_sheets(request, assigned_school)
     else:
-        pass #Error handling?
+        register_result = generate_answer_sheets(request, [assigned_school])
+    answer_sheets_pdf = pisa.pisaDocument(StringIO.StringIO(answer_sheets_html.encode("UTF-8")), answer_sheets_result, encoding='UTF-8')
+    merger = PdfFileMerger()
+    merger.append(register_result)
+    merger.append(answer_sheets_result)
+    merger.append(os.path.join(__file__,"..","..","Declaration","Declaration.pdf"))
+    merger.write(os.path.join(__file__,"..","..","Declaration","temp_answer_sheets.pdf"))
```

Comment:
Oh yeah, I forgot to mention. Can you create a temporary file/dir instead of hardcoding the path. Hard coding the path might cause issues if two requests happen at the same time.

Reply:
Actually, even better, I think just using a `StringIO` here will work.

Reply:
Sure, no worries